frontend/src/tests/users.test.tsx

```diff
@@ -0,0 +1,172 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { expect, vi } from 'vitest'
+import { Page } from '../pages/users/+Page'
+import { type PageContext } from 'vike/types'
+import { AuthContext, AuthStatus } from '../components/useAuthContext'
+import { PageContextProvider } from '../components/usePageContext'
+
+describe('Users page', async () => {
+  it('User is not admin', () => {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const pageContext = {
+      urlPathname: '/users'
+    } as PageContext
+
+    render(
+          <PageContextProvider pageContext={pageContext}>
+        <AuthContext.Provider value={{ status: AuthStatus.SignedIn, isAdmin: false, isManager: false, email: 'rice@rice.edu' }}>
+          <Page>
+          </Page>
+        </AuthContext.Provider>
+        </PageContextProvider>
+    )
+    const pageTitle = screen.getByText('This page is admin only')
+    expect(pageTitle).toBeInTheDocument()
+  })
+  it('fetches users when admin is logged in', async () => {
+    const mockUsers = [
+      { id: 1, email: 'user1@rice.edu', is_admin: true, is_manager: false },
+      { id: 2, email: 'user2@rice.edu', is_admin: false, is_manager: true }
+    ]
+    global.fetch = vi.fn().mockResolvedValue({
+      ok: true,
+      json: async () => await Promise.resolve(mockUsers)
+    })
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const pageContext = {
+      urlPathname: '/users'
+    } as PageContext
+
+    render(
+        <PageContextProvider pageContext={pageContext}>
+          <AuthContext.Provider value={{ status: AuthStatus.SignedIn, isAdmin: true, isManager: false, email: 'admin@rice.edu' }}>
+            <Page />
+          </AuthContext.Provider>
+        </PageContextProvider>
+    )
+
+    // Wait for fetchUsers to be called
+    await screen.findByText('user1@rice.edu')
+    expect(global.fetch).toHaveBeenCalledWith('/api/user')
+    expect(screen.getByText('user1@rice.edu')).toBeInTheDocument()
+    expect(screen.getByText('user2@rice.edu')).toBeInTheDocument()
+  })
+
+  it('adds a user when addUser is called', async () => {
+    global.fetch = vi.fn().mockResolvedValueOnce({
+      ok: true,
```

Comment:
Minor: I'm not immediately sure why, but it looks like coverage isn't showing what I'd expect from this test case :thinking: 

![image](https://github.com/openstax/rope/assets/839025/97f6d8ed-78e0-4ada-9908-5f3fbc417919)